src/smc-webapp/jupyter/main.tsx

```diff
@@ -90,213 +125,332 @@ interface JupyterEditorProps {
-  kernel_selection?: immutable.Map<string, any>;
-  kernels_by_name?: immutable.OrderedMap<string, immutable.Map<string, string>>;
-  kernels_by_language?: immutable.OrderedMap<string, immutable.List<string>>;
-  default_kernel?: string;
-  closestKernel?: KernelType;
-  kernel_streams?: immutable.Map<string, string>;
-  kernel_error?: string;
 }
 
-class JupyterEditor0 extends Component<JupyterEditorProps> {
-  public static reduxProps({ name }) {
-    return {
-      [name]: {
-        kernel: rtypes.string, // string name of the kernel
-        kernels: rtypes.immutable.List,
-        error: rtypes.string,
-        fatal: rtypes.string, // *FATAL* error; user must edit file to fix.
-        toolbar: rtypes.bool,
-        has_unsaved_changes: rtypes.bool,
-        cell_list: rtypes.immutable.List, // list of ids of cells in order
-        cells: rtypes.immutable.Map, // map from ids to cells
-        cm_options: rtypes.immutable.Map, // settings for all the codemirror editors
-        project_id: rtypes.string,
-        directory: rtypes.string,
-        version: rtypes.object,
-        complete: rtypes.immutable.Map, // status of tab completion
-        more_output: rtypes.immutable.Map,
-        about: rtypes.bool,
-        backend_kernel_info: rtypes.immutable.Map,
-        confirm_dialog: rtypes.immutable.Map,
-        find_and_replace: rtypes.bool,
-        keyboard_shortcuts: rtypes.immutable.Map,
-        nbconvert: rtypes.immutable.Map, // backend convert state
-        nbconvert_dialog: rtypes.immutable.Map, // frontend modal dialog state
-        path: rtypes.string,
-        cell_toolbar: rtypes.string,
-        insert_image: rtypes.string, // show insert image dialog
-        edit_attachments: rtypes.string,
-        edit_cell_metadata: rtypes.immutable.Map,
-        metadata: rtypes.immutable.Map,
-        trust: rtypes.bool,
-        kernel_info: rtypes.immutable.Map,
-        check_select_kernel_init: rtypes.bool,
-        show_kernel_selector: rtypes.bool,
-        kernel_selection: rtypes.immutable.Map,
-        kernels_by_name: rtypes.immutable.Map,
-        kernels_by_language: rtypes.immutable.Map,
-        default_kernel: rtypes.string,
-        kernel_error: rtypes.string,
-        closestKernel: rtypes.immutable.Map,
-      },
-      customize: { site_name: rtypes.string },
-      account: { editor_settings: rtypes.immutable.Map },
+export const JupyterEditor: React.FC<Props> = React.memo((props: Props) => {
+  const {
+    actions,
+    editor_actions,
+    frame_actions,
+    name,
+    is_focused,
+    is_fullscreen,
+    font_size,
+    mode,
+    cur_id,
+    sel_ids,
+    md_edit_ids,
+    scroll,
+    scrollTop,
+    hook_offset,
+    view_mode,
+  } = props;
+
+  const site_name = useTypedRedux("customize", "site_name");
+  // const account = useTypedRedux("account", "editor_settings");
+
+  // status of tab completion
+  const complete: undefined | immutable.Map<any, any> = useRedux([
```

Comment:
A todo for me is to make it possible via a hook to do all this without having to explicitly put any types here (it comes from the declared types for the Store).    That's *definitely* what should get done here.  We can't use useEditorRedux as is, because that is for an arbitrary project_id/path, not name, but hopefully a slight modification or extension of that will work.  Not for you to worry about here, but for me...